docs/advanced_topics/customization/page_editing_interface.md

```diff
@@ -292,3 +292,166 @@ class BlogPage(Page):
         FieldPanel('body'),
     ]
 ```
+
+(customizing_slug_widget)=
+
+## Customizing Page slug generation
+
+The `SlugInput` accepts additional kwargs or can be extended for custom slug generation.
+
+Django's `models.SlugField` fields will automatically use the Wagtail admin's `SlugInput`, you will first need to override the widget.
+
+```python
+# ... imports
+
+class MyPage(Page):
+    promote_panels = [
+        FieldPanel("slug"), # automatically uses `SlugInput`
+        # ... other panels
+    ]
+```
+
+### Overriding `SlugInput`
+
+There are multiple ways to override the `SlugInput`, depending on your use case and admin setup.
+
+#### Via `promote_panels`
+
+The simplest, if have already set custom `promote_panels` is to leverage the `FieldPanel` widget kwarg as follows.
+
+```python
+from wagtail.admin.widgets.slug import SlugInput
+# ... other imports
+
+class MyPage(Page):
+    promote_panels = [
+        FieldPanel("slug", widget=SlugInput(locale="uk-UK")), # force a specific locale for this page's slug only
+        # ... other panels need to be declared
+    ]
+```
+
+#### Via a custom form using `base_form_class`
+
+If you do not want to re-declare the `promote_panels`, the `base_form_class` can reference a form class that sets the widget override.
+
+```py
+from wagtail.admin.forms import WagtailAdminPageForm
+from wagtail.admin.widgets import SlugInput
+from wagtail.models import Page
+# ... other imports
+
+
+class MyPageForm(WagtailAdminPageForm):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # force a specific locale for this page's slug only
+        self.fields["slug"].widget = SlugInput(locale="uk-UK")
+
+
+class MyPage(Page):
+    base_form_class = MyPageForm
+    # other fields
+    # no need to declare `promote_panels`
+```
+
+#### Globally, via `register_form_field_override`
+
+If you want to override the `models.SlugField` widget for the entire admin, this can be done using the Wagtail admin util.
+
+It's best to add this to a `wagtail_hook.py` file as this will get run at the right time.
+
+```py
+from django.db import models
+from wagtail.admin.forms.models import register_form_field_override
+from wagtail.admin import widgets
+
+# .. other imports & hooks
+
+
+register_form_field_override(
+    models.SlugField,
+    override={"widget": widgets.SlugInput(locale="uk-UK")},
+)
+```
+
+The following sections will only focus on the `SlugInput(...` usage and not where to override this.
+
+### Overriding the default locale behavior via `locale`
+
+The `SlugInput` is locale aware and will adjust the transliteration (Unicode to ASCII conversion) based on the determined model locale, only when [`ALLOW_UNICODE_SLUGS`](wagtail_allow_unicode_slugs) is `False`.
+
+The locale will be determined from the target translation locale if ()[internationalisation] is enabled, or the current admin user's language if not.
```

Comment:
```suggestion
The locale will be determined from the target translation locale if ()[internationalisation] is enabled. If Internationalization is not in use, it will be based on the language of the admin for the current logged in user. This can be overridden, see below.
```